pkg/storage/etcd/etcd_helper.go

```diff
@@ -40,15 +40,29 @@ import (
 	"golang.org/x/net/context"
 )
 
+// ValueTransformer allows a string value to be transformed before being read from or written to the underlying store.
+type ValueTransformer interface {
+	ReadString(string) (string, error)
+	WriteString(string) (string, error)
+}
+
+type identityTransformer struct{}
+
+func (identityTransformer) ReadString(s string) (string, error)  { return s, nil }
+func (identityTransformer) WriteString(s string) (string, error) { return s, nil }
+
+var IdentityTransformer ValueTransformer = identityTransformer{}
+
 // Creates a new storage interface from the client
 // TODO: deprecate in favor of storage.Config abstraction over time
-func NewEtcdStorage(client etcd.Client, codec runtime.Codec, prefix string, quorum bool, cacheSize int) storage.Interface {
+func NewEtcdStorage(client etcd.Client, codec runtime.Codec, prefix string, quorum bool, cacheSize int, transformer ValueTransformer) storage.Interface {
```

Comment:
Wouldn't it be simpler to decorate `etcdKeysAPI` with the transformation than adding ReadString/WriteString everywhere?